draco/util/tools.py

```diff
@@ -534,3 +539,152 @@ def window_generalised(x, window="nuttall"):
 
     w = (a[:, np.newaxis] * np.cos(t)).sum(axis=0)
     return np.where((x >= 0) & (x <= 1), w, 0)
+
+
+def arPLS_1d(y, mask=None, scale=1, end_frac=1e-2, max_iter=1000):
+    """Use arPLS to estimate a signal baseline.
+
+    Asymmetrically reweighted penalized least squares solve for a signal
+    baseline in the presence of high power outliers by heavily weighting
+    values below the signal and minimizing the weights of values above
+    the signal.
+
```

Comment:
It would be useful to have a brief explanation of what the algorithm is actually doing. This seems to be a good reference: https://pybaselines.readthedocs.io/en/latest/algorithms/whittaker.html

Reply:
Oh cool, I never looked to see if there was an existing python implementation of this